From ff10592d4710f12d601dcfcdd25f28b6941c5141 Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Thu, 24 Mar 2022 03:23:02 -0400
Subject: [PATCH] Remove submodules

Rebase to edk2-stable202311: removing additional submodule:

- CryptoPkg/Library/MbedTlsLib/mbedtls

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>

Rebase to edk2-stable202305: removing additional submodules:

-  MdePkg/Library/BaseFdtLib/libfdt
-  MdePkg/Library/MipiSysTLib/mipisyst
-  UnitTestFrameworkPkg/Library/GoogleTestLib/googletest
-  UnitTestFrameworkPkg/Library/SubhookLib/subhook

Signed-off-by: Oliver Steffen <osteffen@redhat.com>

Upstream edk2 tracks several submodules we do not need in RHEL (removal
done by individual commits in previous RHEL versions):

- openssl: We use RHEL specific openssl submodule later (commit 48f993088e)
- SoftFloat: required only for 32-bit ARM (commit 273787a5c2)
- cmocka: needed for UnitTestFrameworkPkg we do not use (commit a2dca9bcd2)
- oniguruma: rhel do not need this dependency (commit 73f4b42b3a)
- brotli: removed this dependency (commits fcd212ffce, cf62a90767 and ac5782e6ab)
- jansson: we do not depend on JSON parsing or formating (commit c84227659a)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>

MdeModulePkg: remove package-private Brotli include path (RH only)

Notes about the RHEL-8.3/20200603-ca407c7246bf [edk2-stable202005] ->
RHEL-8.5/20210520-e1999b264f1f [edk2-stable202105] rebase:

- no change

Notes about the RHEL-8.2/20190904-37eef91017ad [edk2-stable201908] ->
RHEL-8.3/20200603-ca407c7246bf [edk2-stable202005] rebase:

- New patch.

Originating from upstream commit 58802e02c41b
("MdeModulePkg/BrotliCustomDecompressLib: Make brotli a submodule",
2020-04-16), "MdeModulePkg/MdeModulePkg.dec" contains a package-internal
include path into a Brotli submodule.

The edk2 build system requires such include paths to resolve successfully,
regardless of the firmware platform being built. Because
BrotliCustomDecompressLib is not consumed by any OvmfPkg or ArmVirtPkg
platforms, and we've removed the submodule earlier in this patch set,
remove the include path too.

Signed-off-by: Laszlo Ersek <lersek@redhat.com>
(cherry picked from commit e05e0de713c4a2b8adb6ff9809611f222bfe50ed)
---
 BaseTools/Source/C/GNUmakefile | 1 -
 CryptoPkg/.gitignore           | 1 +
 MdeModulePkg/MdeModulePkg.dec  | 3 ---
 MdePkg/MdePkg.dec              | 5 -----
 4 files changed, 1 insertion(+), 9 deletions(-)
 create mode 100644 CryptoPkg/.gitignore

diff --git a/BaseTools/Source/C/GNUmakefile b/BaseTools/Source/C/GNUmakefile
index 5275f657ef..39d7199753 100644
--- a/BaseTools/Source/C/GNUmakefile
+++ b/BaseTools/Source/C/GNUmakefile
@@ -51,7 +51,6 @@ all: makerootdir subdirs
 LIBRARIES = Common
 VFRAUTOGEN = VfrCompile/VfrLexer.h
 APPLICATIONS = \
-  BrotliCompress \
   VfrCompile \
   EfiRom \
   GenFfs \
diff --git a/CryptoPkg/.gitignore b/CryptoPkg/.gitignore
new file mode 100644
index 0000000000..68b83272b7
--- /dev/null
+++ b/CryptoPkg/.gitignore
@@ -0,0 +1 @@
+Library/OpensslLib/openssl*/
diff --git a/MdeModulePkg/MdeModulePkg.dec b/MdeModulePkg/MdeModulePkg.dec
index d2fede4f87..265dfec94f 100644
--- a/MdeModulePkg/MdeModulePkg.dec
+++ b/MdeModulePkg/MdeModulePkg.dec
@@ -26,9 +26,6 @@
   Include
   Test/Mock/Include
 
-[Includes.Common.Private]
-  Library/BrotliCustomDecompressLib/brotli/c/include
-
 [LibraryClasses]
   ##  @libraryclass  Defines a set of methods to reset whole system.
   ResetSystemLib|Include/Library/ResetSystemLib.h
diff --git a/MdePkg/MdePkg.dec b/MdePkg/MdePkg.dec
index ac54338089..29f0a6e178 100644
--- a/MdePkg/MdePkg.dec
+++ b/MdePkg/MdePkg.dec
@@ -29,7 +29,6 @@
   Include
   Test/UnitTest/Include
   Test/Mock/Include
-  Library/MipiSysTLib/mipisyst/library/include
 
 [Includes.IA32]
   Include/Ia32
@@ -295,10 +294,6 @@
   #
   FdtLib|Include/Library/FdtLib.h
 
-  ##  @libraryclass  Provides general mipi sys-T services.
-  #
-  MipiSysTLib|Include/Library/MipiSysTLib.h
-
   ##  @libraryclass  Provides API to output Trace Hub debug message.
   #
   TraceHubDebugSysTLib|Include/Library/TraceHubDebugSysTLib.h
